frontend/src/features/admin-form/responses/ResponsesPage/storage/UnlockedResponses/UnlockedResponsesProvider.tsx

```diff
@@ -0,0 +1,137 @@
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react'
+
+import { StorageModeSubmissionMetadata, SubmissionId } from '~shared/types'
+
+import { useFormResponses } from '~features/admin-form/responses/queries'
+
+import { usePageSearchParams } from './hooks/usePageSearchParams'
+
+interface UnlockedResponsesContextProps {
+  currentPage?: number
+  setCurrentPage: (page: number) => void
+  count?: number
+  metadata: StorageModeSubmissionMetadata[]
+  isLoading: boolean
+  isAnyLoading: boolean
+  getNextSubmissionId: (currentSubmissionId: string) => SubmissionId | undefined
+  getPreviousSubmissionId: (
+    currentSubmissionId: string,
+  ) => SubmissionId | undefined
+  lastNavPage?: number
+}
+
+const UnlockedResponsesContext = createContext<
+  UnlockedResponsesContextProps | undefined
+>(undefined)
+
+export const useUnlockedResponses = (): UnlockedResponsesContextProps => {
+  const context = useContext(UnlockedResponsesContext)
+  if (!context) {
+    throw new Error(
+      `useUnlockedResponsesContext must be used within a UnlockedResponsesProvider component`,
+    )
+  }
+  return context
+}
+
+const useProvideUnlockedResponses = () => {
+  const [currentPage, setCurrentPage] = usePageSearchParams()
+  // Storing the params in the state for navigation when user returns from
+  // individual response view.
+  const [lastNavPage, setLastNavPage] = useState(currentPage)
+
+  useEffect(() => {
+    if (currentPage && currentPage !== lastNavPage) {
+      setLastNavPage(currentPage)
+    }
+  }, [currentPage, lastNavPage])
+
+  const { data: { count, metadata = [] } = {}, isLoading } =
+    useFormResponses(currentPage)
+  const {
+    data: { metadata: prevMetadata = [] } = {},
+    isLoading: isPrevLoading,
+  } = useFormResponses(currentPage ?? 1 - 1)
```

Comment:
Doesn't this need parentheses to do what you expect? The nullish-coalescing operator has [lower precendence](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence) than substraction, so isn't what's written here incorrectly equivalent to `(currentPage ?? 0)`?

Reply:
yes, we want page 0 in that case. Looking at this i'll simplify the equation so it's not so confusing haha